### PR TITLE
fix: clear device metrics overrides when a DevTools client disconnects

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

When a CDP client sets `Emulation.setDeviceMetricsOverride` and disconnects without clearing it (killed process, dropped websocket), the page stays pinned at the override size: the view keeps its emulated size, and `WebContentsImpl::GetSizeForMainFrame()` re-applies it on every navigation. Only relaunching the app recovers. Root cause is upstream — session teardown (`EmulationHandler::Disable()`) skips the `ClearDeviceEmulationSize()` cleanup that an explicit Emulation.clearDeviceMetricsOverride performs.

This fixes it in Electron's `DevToolsManagerDelegate`, which content notifies for every DevTools client regardless of transport — remote-debugging websocket clients, `webContents.debugger`, the bundled DevTools frontend and extensions all attach, dispatch and detach through the same `DevToolsSession` path, and all of them hit the same missing cleanup in `EmulationHandler::Disable()`. `HandleCommand()` records which client channels hold an uncleared `Emulation.setDeviceMetricsOverride`, and `ClientDetached()` restores the WebContents' device emulation size when such a channel goes away. The bookkeeping is per channel, so a detach only undoes that channel's own override: if another still-attached client holds an override on the same WebContents, the size is left alone until that client clears it or detaches.

Verified affected: 42.5.1, 43.0.0, main, and stock Chromium — an upstream fix in EmulationHandler::Disable() is the long-term home for this. Includes regression specs for both entry points: one sets an override over a raw remote-debugging websocket, severs the connection, and asserts the page returns to the window size; the other does the same through `webContents.debugger.detach()`.

#### Checklist

- [x] I have built and tested this change
- [x] I have filled out the PR description
- [x] [I have reviewed and verified the changes](https://github.com/electron/governance/blob/main/policy/ai.md)
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant API documentation, tutorials, and examples are updated and follow the [documentation style guide](https://github.com/electron/electron/blob/main/docs/development/style-guide.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixed DevTools device metrics overrides persisting indefinitely when a remote debugging client disconnected without clearing them.

### CODE REVIEW

File: shell/browser/ui/devtools_manager_delegate.cc [hunk 1]
```diff
@@ -108,6 +114,14 @@ void DevToolsManagerDelegate::HandleCommand(
...
                                    crdtp::FallthroughCallback());
   DCHECK(dispatchable.ok());
+  if (crdtp::SpanEquals(crdtp::SpanFrom(kSetDeviceMetricsOverrideMethod),
+                        dispatchable.Method())) {
+    channels_with_device_overrides_.insert(channel);
+  } else if (crdtp::SpanEquals(
+                 crdtp::SpanFrom(kClearDeviceMetricsOverrideMethod),
+                 dispatchable.Method())) {
+    channels_with_device_overrides_.erase(channel);
+  }
```
Comment (deepak1556) [inline shell/browser/ui/devtools_manager_delegate.cc:120]: This is not filtering to remote debugging sessions as the PR body claims. Does this issue only affect remote sessions ? 

>  Overrides owned by other sessions (built-in DevTools, webContents.debugger) stay active.

Isn't the manager same for all sessions. If you want to guard for all sessions then current change is good otherwise we need a session filter here. I couldn't find a real type differentiator, the closest is `DevToolsAgentHostClient::GetTypeForMetrics`.

**Comment by MarshallOfSound** [comment]
Heads up: Windows CI builds moved to new runners on Monday and the old ones have now been removed. This branch predates that, so its Windows build jobs still point at the old runners and will sit queued forever if CI runs again. Rebasing on `main` (or "Update branch") picks up the change, nothing else needed.

File: shell/browser/ui/devtools_manager_delegate.cc (hunk 1 above)
Comment (codebytere) [reply shell/browser/ui/devtools_manager_delegate.cc:120]: not remote-only, the body was badly worded (fixed now). every client type (remote socket, `webContents.debugger`, bundled frontend, extensions) goes through the same `DevToolsSession` attach/dispatch/detach path and hits the same missing `ClearDeviceEmulationSize()` in `EmulationHandler::Disable()`, and `HandleCommand()`/`ClientDetached()` fire for all of them, so guarding every session is intentional and no type filter is needed.

what "other sessions stay active" meant is that the bookkeeping is per channel: a detach only undoes an override that channel set. latest push makes that strictly true, so if another attached client still holds an override on the same `WebContents` we leave the size alone until it clears or detaches. also added a `webContents.debugger.detach()` spec next to the websocket one so both entry points are covered.

**Review by MarshallOfSound — APPROVED** [review]

**Review by VerteDinde — APPROVED** [review]

**Review by deepak1556 — APPROVED** [review]
